ArduCopter/events.cpp

```diff
@@ -4,152 +4,235 @@
  *       This event will be called when the failsafe changes
  *       boolean failsafe reflects the current state
  */
+
+bool Copter::failsafe_option(FailsafeOption opt) const
+{
+    return (g2.fs_options & (uint32_t)opt);
+}
+
 void Copter::failsafe_radio_on_event()
 {
-    // if motors are not armed there is nothing to do
-    if( !motors->armed() ) {
-        return;
-    }
+    AP::logger().Write_Error(LogErrorSubsystem::FAILSAFE_RADIO, LogErrorCode::FAILSAFE_OCCURRED);
 
+    int16_t action;
+    action = g.failsafe_throttle; //The action set in FS_THR_ENABLE
+
+    // Conditions to deviate from FS_THR_ENABLE selection and sent specigic GCS warning
     if (should_disarm_on_failsafe()) {
+        // should immediately disarm when we're on the ground
+        gcs().send_text(MAV_SEVERITY_WARNING, "RADIO FAILSAFE - DISARMING");
```

Comment:
I wonder if we really want the extra send_texts?  MP already has a pretty easy to read display.  In general I prefer we limit the use of send_text but this isn't a blocker if people think it's really necessary.

Reply:
The GCS tells you that you're in failsafe, but not why.  Now with these options, IMO it is important for the operator to be aware of what the copter is doing.